### PR TITLE
Use pfdhcp API to perform mac2ip + ip2mac

### LINKED ISSUE #3133: DHCP: mac2ip only uses the DB tables
We used to have the OMAPI for faster lookups via ISC DHCP, the new DHCP server has an API but its not being used by mac2ip and potentially ip2mac (needs confirmation)

We should implement something similar to the pf::OMAPI module with a cache as L1 layer and call the DHCP API in cache misses

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Allows to use the pfdhcp API for ip/mac lookups similarly to what we were doing with the OMAPI

# Impacts
IP/MAC lookups

# Issue
fixes #3133 

# Delete branch after merge
YES

### CODE REVIEW

File: go/dhcp/api.go [hunk 1]
```diff
@@ -198,13 +196,9 @@ func handleReleaseIP(res http.ResponseWriter, req *http.Request) {
 	vars := mux.Vars(req)
 	_ = InterfaceScopeFromMac(vars["mac"])
 
-	var result = map[string][]*Info{
-		"result": {
-			&Info{Mac: vars["mac"], Status: "ACK"},
-		},
-	}
+	var result = &Info{Mac: vars["mac"], Status: "ACK"},
```
Comment (fdurand) [inline go/dhcp/api.go]: Remove the comma at the end

File: go/dhcp/api.go [hunk 2]
```diff
@@ -198,13 +196,9 @@ func handleReleaseIP(res http.ResponseWriter, req *http.Request) {
...
 
-	res.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		res.Header().Set("Content-Type", "application/json; charset=UTF-8")
```
Comment (fdurand) [inline go/dhcp/api.go]: Remove 1 tab

File: go/dhcp/api.go [hunk 3]
```diff
@@ -254,13 +244,9 @@ func handleOverrideNetworkOptions(res http.ResponseWriter, req *http.Request) {
...
 
-	var result = map[string][]*Info{
-		"result": {
-			&Info{Network: vars["network"], Status: "ACK"},
-		},
-	}
+	var result = &Info{Network: vars["network"], Status: "ACK"},
 
-	res.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		res.Header().Set("Content-Type", "application/json; charset=UTF-8")
```
Comment (fdurand) [inline go/dhcp/api.go]: Remove 1 tab

File: go/dhcp/api.go [hunk 4]
```diff
@@ -254,13 +244,9 @@ func handleOverrideNetworkOptions(res http.ResponseWriter, req *http.Request) {
 	// Insert information in etcd
 	_ = etcdInsert(vars["network"], sharedutils.ConvertToString(body))
 
-	var result = map[string][]*Info{
-		"result": {
-			&Info{Network: vars["network"], Status: "ACK"},
-		},
-	}
+	var result = &Info{Network: vars["network"], Status: "ACK"},
```
Comment (fdurand) [inline go/dhcp/api.go]: Remove the comma at the end

**Review by fdurand — CHANGES_REQUESTED** [review]

File: lib/pf/dhcp/api.pm [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+Allows to access the pfdhcp API through the Unified API
+
+=head1 SYNOPSIS
+
+    use pf::dhcp::api;
+
+    my $dhcp_api = pf::dhcp::api->new(host => 'localhost');
+
+    my $data = $dhcp_api->lookup({'ip-address' => "10.229.25.247" });
```
Comment (julsemaan) [inline lib/pf/dhcp/api.pm]: Note to self:

> You are wrong

**Comment by julsemaan** [comment]
Completed adjustments

**Review by fdurand — APPROVED** [review]

**Pull request merged**